payment-manager/src/test/java/com/qpay/paymentmanager/service/impl/WalletServiceImplTest.java

```diff
@@ -109,4 +112,173 @@ void should_deleteWallet() {
         //then
         then(walletRepository).should().deleteById(1L);
     }
+
+    @Test
+    void should_makePaymentBetweenWallets() {
+        // when
+        var fromId = 1L;
+        var toId = 2L;
+        var walletPayment = WalletPayment.builder()
+                .walletIdFrom(fromId)
+                .walletIdTo(toId)
+                .amount(BigDecimal.valueOf(100))
+                .build();
+
+        var fromWallet = WalletEntity.builder()
+                .id(fromId)
+                .name("CustomerWallet")
+                .balance(new BigDecimal(200))
+                .userId(1L)
+                .userType(UserType.CUSTOMER)
+                .build();
+
+        var toWallet = WalletEntity.builder()
+                .id(toId)
+                .name("MerchantWallet")
+                .balance(new BigDecimal(300))
+                .userId(2L)
+                .userType(UserType.CUSTOMER)
+                .build();
+
+        var expectedFromWallet = WalletEntity.builder()
+                .id(fromId)
+                .name("CustomerWallet")
+                .balance(new BigDecimal(100))
+                .userId(1L)
+                .userType(UserType.CUSTOMER)
+                .build();
+
+        var expectedToWallet = WalletEntity.builder()
+                .id(toId)
+                .name("MerchantWallet")
+                .balance(new BigDecimal(400))
+                .userId(2L)
+                .userType(UserType.CUSTOMER)
+                .build();
+
+        given(walletRepository.findById(fromId)).willReturn(Optional.of(fromWallet));
+        given(walletRepository.findById(toId)).willReturn(Optional.of(toWallet));
+        given(walletRepository.save(expectedFromWallet)).willReturn(expectedFromWallet);
+        given(walletRepository.save(expectedToWallet)).willReturn(expectedToWallet);
+
+        // when
+        var result = walletService.makePayment(walletPayment);
+
+        // then
+        assertThat(result).isEqualTo(expectedFromWallet);
+    }
+
+    @Test
+    void should_throwNoWalletFoundException_when_providePaymentWalletDoesntExistWithIdFrom() {
```

Comment:
```suggestion
    void should_throwNoWalletFoundException_when_walletDoesntExistByIdFrom() {
```

and make other tests correctly from English point of view